File: sources/python/fib.py
# http://www.balancer.ru/g/p1618868

class Fib:
    def __init__(self, n):
        self._value = n

    def value(self):
        if(self._value <= 2):
            return 1

        f1 = Fib(self._value - 1)
        f2 = Fib(self._value - 2)

        return f1.value() + f2.value()

print Fib(40).value()

File: sources/dart/fib.dart
class Fib
{
  int n;
  Fib(this.n);
  int value()
  {
    if(n <= 2 )
    {
      return 1;
    }
    Fib f1 = new Fib(n-1);
    Fib f2 = new Fib(n-2);
    return f1.value() + f2.value();
  }
}

void main() {
  print(new Fib(40).value());
}
